### ISSUE TITLE
Typescript setup without sveltekit

### ISSUE THREAD

**Issue by gwendalF** [bryanmylee/svelte-headless-table#150]
Hello, I have the same issue as #https://github.com/bryanmylee/svelte-headless-table/issues/60  with typings using npm. I have the version 0.17.5
I don't use sveltekit but only svelte. I tried with sveltekit and it works

I think there is an issue on how the types are exported and mapped

**Comment by bryanmylee** [comment]
SvelteKit is probably exporting the types weird...

**Comment by lukeed** [comment]
It's the package itself, nothing to do with SvelteKit. 

Recent TypeScript versions added new `moduleResolution` values (including new default). The most common value is `bundler` which will tell TS to completely ignore the `"types"` field that you have listed for this package. Instead, `bundler` will **only** traverse an `"exports"` map for type definitions. See https://github.com/bryanmylee/svelte-headless-table/issues/60#issuecomment-1680237133 for tiny working example.

You'd need a `exports` map entry for each entry of the module – which i think is root (`.`) and `./plugins`... which then means you also need a `"types"` condition for each of those entries. Something like:

```js
"exports": {
  ".": {
    "types": "./lib/index.d.ts",
    "default": "./index.js"
  },
  "./plugins": {
    "types": "./lib/plugins.d.ts",
    "default": "./plugins/index.js"
  }
}
```

**Comment by bryanmylee** [comment]
Fixed by #178 and #181. I'll be releasing a v1 to cover this.
